src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -545,6 +551,92 @@ emit_sum_vector (MonoCompile *cfg, MonoType *vector_type, MonoTypeEnum element_t
 }
 #endif
 
+#ifdef TARGET_AMD64
+static int type_to_extract_op (MonoTypeEnum type);
+static const int fast_log2 [] = { 1, 0, 1, -1, 2, -1, -1, -1, 3 };
```

Comment:
This is not simply calculating log2. It seems that you've assigned `-1` to places where you think should be illegal element numbers. If that's the case, element number 0 and 1 should be `-1` as well.